proposals/NNNN-async-stream-backpressure.md

```diff
@@ -0,0 +1,784 @@
+# Backpressure support for AsyncStream
+
+* Proposal: [SE-NNNN](NNNN-async-stream-backpressure.md)
+* Authors: [Franz Busch](https://github.com/FranzBusch)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/swift#66488](https://github.com/apple/swift/pull/66488)
+* Review: ([pitch](https://forums.swift.org/t/pitch-new-apis-for-async-throwing-stream-with-backpressure-support/65449))
+
+## Introduction
+
+[SE-0314](https://github.com/apple/swift-evolution/blob/main/proposals/0314-async-stream.md)
+introduced new `Async[Throwing]Stream` types which act as root asynchronous
+sequences. These two types allow bridging from synchronous callbacks such as
+delegates to an asynchronous sequence. This proposal adds a new way of
+constructing asynchronous streams with the goal to bridge backpressured systems
+into an asynchronous sequence.
```

Comment:
Perhaps salient here to include something, as @weissi commented in the pitch thread, about also addressing shortcomings in consumer-to-producer and producer-to-consumer cancellation: seems like that's an important part of this proposal worth calling out upfront.